src/api/index.ts

```diff
@@ -0,0 +1,108 @@
+import express from 'express';
+import { body, validationResult } from 'express-validator';
+import { PrismaClient } from '@prisma/client';
+import logger from '../logger';
+
+const prisma = new PrismaClient();
+const router = express.Router();
+
+router.get('/', async (req, res) => {
+  return res.json({ msg: 'The API is working' });
+});
+
+/**
+ * @description creates a subscription for a deviceToken adding the provided wallet address
+ * @param {string} deviceToken - the device token of user's device (e.g. phone)
+ * @param {string} walletAddress - the walletAddress of a user
+ */
+router.post('/subscriptions', body('deviceToken').notEmpty(), body('walletAddress').notEmpty(), async (req, res) => {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array() });
+  }
+
+  const { deviceToken, walletAddress } = req.body;
+
+  let subscription = await prisma.subscription.findUnique({
+    where: {
+      deviceToken: deviceToken,
+    },
+    include: {
+      walletAddresses: true,
+    },
+  });
+
+  if (!subscription) {
+    subscription = await prisma.subscription.create({
+      data: {
+        deviceToken,
+        walletAddresses: {
+          create: [{ address: walletAddress }],
+        },
+      },
+      include: {
```

Comment:
How to understand this?

Reply:
When creating an entry in the db prisma returns the data stored and since walletAddress is a relation (other table) this includes the address in the query